client/app/components/queries/query-editor.js

```diff
@@ -113,6 +114,10 @@ function queryEditor(QuerySnippet, $timeout) {
               editor.resize();
             });
 
+            $scope.$parent.$watch('autoCompleteQuery', () => {
+              editor.setOption('enableLiveAutocompletion', $scope.$parent.autoCompleteQuery);
+            });
```

Comment:
It's better to pass that option as component attribute binding and not access to parent scope

Reply:
For this component: go to `scope: { ... }` definition object and add line: `autoCompleteQuery: '='` ([here](https://github.com/getredash/redash/blob/e2b41ebbb3dd36102b29381118f95522422a9c59/client/app/components/queries/query-editor.js#L28)). Then use it as `$scope.autoCompleteQuery` instead of `$scope.$parent.autoCompleteQuery`.

Then, [here](https://github.com/getredash/redash/blob/e2b41ebbb3dd36102b29381118f95522422a9c59/client/app/pages/queries/query.html#L146) add new attribute: `auto-complete-query="autoCompleteQuery"` (the variable from "parent" scope).

Reply:
Thank you so much, that makes much more sense.